[WIP][HOTFIX] Fix module problems of mv and spark with spark binary version

ajantha-bhat opened the pull request:
### Why is this PR needed?
 After #3700 , spark binary version addition. 
a) Module name always taking deafult profile spark binary version.
b)   IDE test cases are running with spark 2.3 even for spark2.4 because of above problem
 
 ### What changes were proposed in this PR?
a)  keep a simple module name without binary version
b) Use the package option with jar file name to have binary version in the final jar. 
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

ajantha-bhat commented:
@kunal642 , @akashrn5 : check it

CarbonDataQA1 commented:
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1492/

CarbonDataQA1 commented:
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3219/

ajantha-bhat commented:
retest this please

CarbonDataQA1 commented:
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1494/

CarbonDataQA1 commented:
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3221/

QiangCai commented:
finalName is ${artifactId}-${version} by default, this change will not impact artifactId and version. 
Other modules will can not find the dependency: carbondata-spark_2.3 and carbondata-spark_2.4.

actually, if you change spark.binary.version to 2.4 in pom.xml of the parent module, IDEA will work again for spark 2.4.

ajantha-bhat commented:
@QiangCai : Developer should not manually modify pom to make it work for spark2.4.

so I fixed like above. 

But some testcase failed to find CSV file after this change. so, I stopped it. Need to analyze why CSV files unable to find because of my change.

ajantha-bhat commented:
retest this please

CarbonDataQA1 commented:
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1660/

CarbonDataQA1 commented:
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3403/

QiangCai commented:
@ajantha-bhat 
1. remove CarbonData jars from your local maven repo at first. 
2. build it with -o, you will find dependency error( can not find the dependency: carbondata-spark_2.3 and carbondata-spark_2.4).

CarbonDataQA1 commented:
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4794/

CarbonDataQA1 commented:
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/3036/

ajantha-bhat commented:
@QiangCai : I think it was failing because I have changed Artifact name without binary version but other module pom file dependency was still using binary version in artifcat id. It clicked me today. I have fixed it. 
You can check the PR @QiangCai , @kunal642

ajantha-bhat commented:
With this PR developer need not have to do manual pom change for compiling 2.4 profile and testcase of spark 2.4 can use spark 2.4  without any manual change

kunal642 commented:
How will an external application specify spark version if they add carbondata-spark dependency?

ajantha-bhat commented:
> How will an external application specify spark version if they add carbondata-spark dependency?
@kunal642 : Yeah, it cannot be added. Now only benefit from this change is jar name is proper. I will move to WIP and check if I get any other solution.

CarbonDataQA1 commented:
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4796/

CarbonDataQA1 commented:
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/3038/
